src/sudoku.py

```diff
@@ -23,12 +23,18 @@ class Sudoku:
     def __init__(self, grid: Grid) -> None:
         if not is_valid_grid(grid):
             raise ValueError('Incorrect grid. Expected grid 9x9')
-        self.__rows = [[Cell(item) for item in row] for row in grid]
+
+        def set_flag_true(self):
+            print("set_flag_true")
+            self.__any_alternatives_excluded = True
```

Comment:
Это не метод а именно функция. 
Методы определяются на уровне класса. А уже тело метода подчиняется тем же законам, что тело модуля или функции. Например, внутри метода можно определить новый класс, с методом внутри:
```python
class A:
    def method_a(self):
        A_name = self.__class__.__name__  # "A" 
        class B:
            def method_b(self):
                print(self.__class__.__name__ + A_name)  # "BA"
```
Но метод внутри метода определить нельзя, только функцию. К счастью, область имён функции **вложена** в область имён метода. Поэтому ссылка на объект `self` тоже доступна:
```suggestion
        def set_progress_flag():
            self.__any_alternatives_excluded = True
```